### PR TITLE
MINOR: Update LICENSE-binary for 3.7.0

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: licenses/protobuf-java-BSD-3-clause [hunk 1]
```diff
@@ -0,0 +1,34 @@
+Copyright 2023, the original author or authors.
```
Comment (mimaison) [inline licenses/protobuf-java-BSD-3-clause:1]: Google does not have a BSD-3-lcause license in [their repo](https://github.com/protocolbuffers/protobuf/tree/main/java) but it seems [this library is still under that license](https://github.com/protocolbuffers/protobuf/blob/main/java/pom.xml#L44-L45). So I built this file from the license template and picked 2023 as the date.

**Comment by mimaison** [comment]
cc @stanislavkozlovski I've have to update `license-BINARY` for 3.6.1 so making the changes for trunk too.

**Review by jlprat — APPROVED** [review]
LGTM thanks @mimaison 
We would need to update the NOTICE-binary as well (as there are a couple of new binaries). I'll get to it

**Comment by mimaison** [comment]
> We would need to update the NOTICE-binary as well (as there are a couple of new binaries). I'll get to it
Ok, thanks!

**Comment by jlprat** [comment]
Created https://github.com/apache/kafka/pull/14865 for the `NOTICE-binary`. In the end, of the new dependencies, only `commons-io` had a `NOTICE`.

**Pull request merged**